api/src/main/java/org/ehcache/spi/serialization/Serializer.java

```diff
@@ -28,8 +28,9 @@
  * <dl>
  *   <dt><code><i>Serializer</i>(ClassLoader loader)</code>
  *   <dd>This constructor is used to initialize the serializer for transient caches.
- *   <dt><code><i>Serializer</i>(ClassLoader loader, org.ehcache.core.spi.service.FileBasedPersistenceContext context)</code>
- *   <dd>This constructor is used to initialize the serializer for persistent caches.
+ *   <dt><code><i>Serializer</i>(ClassLoader loader, org.ehcache.spi.persistence.StateRepository stateRepository)</code>
+ *   <dd>This constructor is used to initialize the serializer for persistent caches and allows them to store any relevant
```

Comment:
No, it is the constructor we use when a persistent tier is configured. Which is what makes this implicit constructor contract even more weird. A statelessl serializer still needs to have this constructor if it is fit for persistent caches.